test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,127 @@
-                            + " -Djava.security.manager",
-                    "TempDirDoesNotExist", "nio");
+    public static Stream<Arguments> noTempDirSource() {
+        return Stream.of(Arguments.of(List.of("TempDirDoesNotExist", "io")),
+                         Arguments.of(List.of("TempDirDoesNotExist", "nio")),
+                         Arguments.of(List.of("-Djava.io.tmpdir=" + USER_DIR,
+                                              "TempDirDoesNotExist", "io")),
+                         Arguments.of(List.of("-Djava.io.tmpdir=" + USER_DIR,
+                                              "TempDirDoesNotExist", "nio")));
+    }
 
-            // error message printed only once
-            testMessageCounter(0, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "io", "nio");
-        }
+    public static Stream<Arguments> counterSource() {
+        // standard test with default setting for java.io.tmpdir
+        return Stream.of(Arguments.of(List.of("-Djava.io.tmpdir=" + tempDir(),
+                                             "TempDirDoesNotExist",
+                                             "io", "nio")));
     }
 
-    private static void testMessageExist(int exitValue, String errorMsg, String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldContain(errorMsg)
-                .shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("tempDirSource")
+    public void existingMessage(List<String> options) throws Exception {
+       ProcessTools.executeTestJvm(options).shouldContain(WARNING)
+           .shouldHaveExitValue(0);
     }
 
-    private static void testMessageNotExist(int exitValue, String errorMsg,String... options) throws Exception {
-        ProcessTools.executeTestJvm(options).shouldNotContain(errorMsg).shouldHaveExitValue(exitValue);
+    @ParameterizedTest
+    @MethodSource("noTempDirSource")
+    public void nonexistentMessage(List<String> options) throws Exception {
+        ProcessTools.executeTestJvm(options).shouldNotContain(WARNING)
+            .shouldHaveExitValue(0);
     }
 
-    private static void testMessageCounter(int exitValue,String... options) throws Exception {
+    @ParameterizedTest
+    @MethodSource("counterSource")
+    public void messageCounter(List<String> options) throws Exception {
         OutputAnalyzer originalOutput = ProcessTools.executeTestJvm(options);
         List<String> list = originalOutput.asLines().stream().filter(line
-                -> line.equalsIgnoreCase(ioWarningMsg)).collect(Collectors.toList());
-        if (list.size() != 1 || originalOutput.getExitValue() != exitValue)
-            throw new Exception("counter of messages is not one, but " + list.size()
-                    + "\n" + originalOutput.asLines().toString() + "\n");
+                -> line.equalsIgnoreCase(WARNING)).toList();
+        if (list.size() != 1)
```

Comment:
Use assertEquals

Reply:
Fixed in cc4a3369871d8ad66162133e51bf625dfcf9b62c.